### PR TITLE
Fix `deploy_call_smart_contract`

### PR DESCRIPTION
**Pull request by akashin**
This is resolving one of the items from https://github.com/near/nearcore/issues/11031

It is quite bizarre that this test is failing in the first place, as the change in the gas costs should be strictly beneficial to the users.
Here is an example failure: https://nayduck.nearone.org/#/test/12357

@Ekleog-NEAR , any ideas?

### CODE REVIEW

**Review by Ekleog-NEAR — APPROVED** [review]
What really surprises me is, the contract originally had an attached gas of 100Ggas. So… I’m not sure I understand how that could have ever worked at all? Our fees before were much higher than that.

Unfortunately AFAIU we can’t really debug nayduck, so I’ll approve so you can land the PR and then we can check if this fixes nayduck.

**Comment by akashin** [comment]
Here are the logs before and after gas cost change: https://gist.github.com/akashin/26eeb3b1d5227a266caa626dddd7337d

TL;DR: I think we broke the guarantee of not increasing the gas usage of contract function calls.

Our change to the gas costs decreased the cost of transactions overall:
- 242795006714200000000, tokens_burnt before
- 30808856714200000000,  tokens_burnt after
It's around 8x decrease in total.

However, these tokens are burnt in two stages:
1. When converting transaction to receipt
  - This is where the function call send and execution cost are charged
2. When executing a receipt with a function call
  - This is where the contract loading gas usage is charged

Stage 1 burns tokens directly. Stage 2 burns them from the attached gas.

For stage 2, specifically, contract loading gas usage increased as expected:
-  21451097250, gas_used for contract loading before
- 107804258265, gas_used for contract loading after
It's a ~85GGas increase.

Now the question is, how many contracts in the wild have actually been broken?
